DCOS/fluentd-testing/fluentd.Tests.ps1

```diff
@@ -0,0 +1,61 @@
+﻿
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+$newConfigPath = Join-Path $here td-agent.conf
+
+function Get-ServiceConfPath() {
+    $serviceInfo = Get-ItemProperty -PATH HKLM:\SYSTEM\CurrentControlSet\Services\fluentdwinsvc
+    $confPath = $serviceInfo.fluentdopt -split " " | where {$_ -like "*.conf"}
+    return $confPath
+}
+
+Describe "Fluentd sanity check" {
+
+    It "Service can be found" {
+        $service = Get-Service "fluentdwinsvc"
+        $service | Should -Not -Be $null
+    }
+
+    It "Service is running" {
+        $service = Get-Service "fluentdwinsvc"
+        $service.Status | Should -Be "Running"
+    }
+
+    It "Can find custom config in current directory" {
+        Test-Path $newConfigPath -PathType Leaf | Should -Be $true
+    }
+
+    It "Can get get service opts" {
+        Get-ServiceConfPath | Should -Not -Be $null
+    }
+}
+
+
+Describe "Fluentd logging" {
+
+    It "Can modify conf file" {
+        Copy-Item $newConfigPath $(Get-ServiceConfPath) -force
+        $LASTEXITCODE | Should -Be 0
+    }
+
+    It "Can restart service to update config" {
+        # Requires Admin rights
+        Restart-Service -Name fluentdwinsvc
+        $service = Get-Service fluentdwinsvc
+        $service.Status | Should -Be "Running"
+    }
+
+    It "Can see changes to file in dynamic directory" {
+        [String] $DateStamp = get-date -Format yyyyMMddTHHmmss
+        $folderpath = "$env:SystemDrive/DCOS/mesos/$DateStamp"
+        $folder = md $folderpath
+        "Hello at $DateStamp" | Out-File -FilePath "$folderpath/stdout" -Encoding ascii
+
+        # Leave time for fluentd to detect change
+        Sleep 2 
```

Comment:
Are `Sleep 2` enough for fluentd to detect changes ?

I would recommend that we poll the fluentd output file for about 30 seconds and if there's any change there, we consider it a failure.

It's safer than sleeping 2 seconds and then see if the change is propagated.